src/context.cc

```diff
@@ -331,9 +331,15 @@ bool ContextBase::onStart(std::shared_ptr<PluginBase> plugin) {
 }
 
 bool ContextBase::onConfigure(std::shared_ptr<PluginBase> plugin) {
+  if (!in_vm_context_created_ && wasm_->on_context_create_) {
+    wasm_->on_context_create_(this, id_, 0);
+    in_vm_context_created_ = true;
+  }
+
   if (isFailed() || !wasm_->on_configure_) {
     return true;
   }
+
   DeferAfterCallActions actions(this);
   plugin_ = plugin;
   auto result =
```

Comment:
Actually, this needs a few changes:

1) `isFailed()` must be checked first, otherwise you might call `wasm_->on_context_create_()` on a failed VM.
2) You should defer the call in case of re-entrant execution by adding `DeferAfterCallActions actions(this);` before `wasm->on_context_create_()`. I think you need to add a guard for the second `DeferAfterCallActions`, so that only one is executed.
3) `in_vm_context_created_ = true` should be marked even if `wasm_->on_context_create_()` isn't exported, to match `onCreate()`.

Reply:
thanks! PTAL